middle_end/flambda2/simplify/number_adjuncts.ml

```diff
@@ -152,20 +152,23 @@ let with_shift shift if_undefined f ~integer_bit_width =
   | Some shift ->
     if shift < 0 || shift >= integer_bit_width then if_undefined else f shift
 
+let compare_unsigned_generic n1 n2 ~compare ~strictly_negative =
+  (* CR-someday mshinwell: Use faster implementation and/or implementation in
+     the stdlib when available. *)
+  match strictly_negative n1, strictly_negative n2 with
+  | true, true -> compare n2 n1
+  | true, false -> -1
+  | false, true -> 1
+  | false, false -> compare n1 n2
```

Comment:
What is this supposed to compute exactly ? I'm expecting `compare -1 -2` to return the same result as `compare_unsigned -1 -2`, not the reverse. And `compare_unsigned -1 0` should return `1` (as `-1` is `max_int` when seen as unsigned), not `-1`.